src/core/jit/input_resolver.rs

```diff
@@ -66,4 +100,79 @@ where
+        &self,
+        parent_name: &str,
+        arg_name: &str,
+        type_of: &Type,
+        default_value: &Option<Output>,
+        value: Option<Output>,
+    ) -> Result<Option<Output>, ResolveInputError> {
+        let is_value_null = value.as_ref().map(|val| val.is_null()).unwrap_or(true);
+        let value: Result<Option<Output>, ResolveInputError> =
+            if !type_of.is_nullable() && value.is_none() {
+                let default_value = default_value.clone();
+                match default_value {
+                    Some(value) => Ok(Some(value)),
+                    None => Err(ResolveInputError::ArgumentIsRequired {
+                        arg_name: arg_name.to_string(),
+                        field_name: parent_name.to_string(),
+                    }),
+                }
+            } else if !type_of.is_nullable() && is_value_null {
+                Err(ResolveInputError::ArgumentIsRequired {
+                    arg_name: arg_name.to_string(),
+                    field_name: parent_name.to_string(),
+                })
+            } else if value.is_none() {
+                let default_value = default_value.clone();
+                Ok(default_value)
+            } else {
+                Ok(value)
+            };
+
+        match value? {
+            Some(mut value) => match self.plan.index.get_input_type_definition(type_of.name()) {
+                Some(def) => {
+                    if let Some(obj) = value.as_object_mut() {
+                        for arg_field in &def.fields {
+                            let parent_name = format!("{}.{}", parent_name, arg_name);
+                            let field_value = obj.get_key(&arg_field.name).cloned();
+                            let field_default = arg_field.default_value.clone();
+                            let field_default = Output::from_json(field_default);
+                            let value = self.recursive_parse_arg(
+                                &parent_name,
+                                &arg_field.name,
+                                &arg_field.of_type,
+                                &field_default,
+                                field_value,
+                            )?;
+                            if let Some(value) = value {
+                                obj.insert_key(&arg_field.name, value);
+                            }
+                        }
+                    } else if let Some(arr) = value.as_array_mut() {
+                        for (index, item) in arr.iter_mut().enumerate() {
+                            let parent_name = format!("{}.{}.{}", parent_name, arg_name, index);
+
+                            *item = self
+                                .recursive_parse_arg(
+                                    &parent_name,
+                                    &index.to_string(),
+                                    type_of,
+                                    default_value,
```

Comment:
is default value is even applicable for the arrays? we can only define defaults only for argument or field in input type. Should be `None`?

Reply:
Good catch, its always None